spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWriteBuilder.java

```diff
@@ -112,42 +136,74 @@ public BatchWrite buildForBatch() {
+      distribution = buildRequiredDistribution();
+      ordering = buildRequiredOrdering(distribution);
     } else {
-      return write.asBatchAppend();
+      LOG.warn("Can't request distribution/ordering as extensions are disabled and spec has non-identity transforms");
+      distribution = Distributions.unspecified();
+      ordering = NO_ORDERING;
     }
-  }
-
-  @Override
-  public StreamingWrite buildForStreaming() {
-    // Validate
-    Preconditions.checkArgument(handleTimestampWithoutZone || !SparkUtil.hasTimestampWithoutZone(table.schema()),
-        SparkUtil.TIMESTAMP_WITHOUT_TIMEZONE_ERROR);
-
-    Schema writeSchema = SparkSchemaUtil.convert(table.schema(), dsSchema);
-    TypeUtil.validateWriteSchema(table.schema(), writeSchema, writeConf.checkNullability(), writeConf.checkOrdering());
-    SparkUtil.validatePartitionTransforms(table.spec());
 
-    // Change to streaming write if it is just append
-    Preconditions.checkState(!overwriteDynamic,
-        "Unsupported streaming operation: dynamic partition overwrite");
-    Preconditions.checkState(!overwriteByFilter || overwriteExpr == Expressions.alwaysTrue(),
-        "Unsupported streaming operation: overwrite by filter: %s", overwriteExpr);
+    return new SparkWrite(spark, table, writeConf, writeInfo, appId, writeSchema, dsSchema, distribution, ordering) {
+
+      @Override
+      public BatchWrite toBatch() {
+        if (rewrittenFileSetId != null) {
+          return asRewrite(rewrittenFileSetId);
+        } else if (overwriteByFilter) {
+          return asOverwriteByFilter(overwriteExpr);
+        } else if (overwriteDynamic) {
+          return asDynamicOverwrite();
+        } else if (overwriteFiles) {
+          return asCopyOnWriteMergeWrite(mergeScan, isolationLevel);
+        } else {
+          return asBatchAppend();
+        }
+      }
+
+      @Override
+      public StreamingWrite toStreaming() {
+        Preconditions.checkState(!overwriteDynamic,
+            "Unsupported streaming operation: dynamic partition overwrite");
+        Preconditions.checkState(!overwriteByFilter || overwriteExpr == Expressions.alwaysTrue(),
+            "Unsupported streaming operation: overwrite by filter: %s", overwriteExpr);
+        Preconditions.checkState(rewrittenFileSetId == null,
+            "Unsupported streaming operation: rewrite");
+
+        if (overwriteByFilter) {
+          return asStreamingOverwrite();
+        } else {
+          return asStreamingAppend();
+        }
+      }
+    };
+  }
 
-    // Get application id
-    String appId = spark.sparkContext().applicationId();
+  private Distribution buildRequiredDistribution() {
+    if (overwriteFiles) {
+      throw new UnsupportedOperationException("Copy-on-write operations are temporarily not supported");
+    } else {
+      DistributionMode distributionMode = writeConf.distributionMode();
+      return SparkDistributionAndOrderingUtil.buildRequiredDistribution(table, distributionMode);
```

Comment:
I think that if mode is `range`, this will not apply `ignoreSortOrder`.

Reply:
No longer applies after the rename.